From fa37813401ff52d78591c262d6542e4d5d935584 Mon Sep 17 00:00:00 2001
From: Andy Gospodarek <andy@greyhouse.net>
Date: Tue, 29 Jun 2010 18:28:12 +0000
Subject: [PATCH] ixgbe: fix panic when shutting down system with WoL enabled
Patch-mainline: v2.6.35, accepted into net-2.6
Git-commit: fa37813401ff52d78591c262d6542e4d5d935584 

This patch added to 2.6.34:

	commit 5f6c01819979afbfec7e0b15fe52371b8eed87e8
	Author: Jesse Brandeburg <jesse.brandeburg@intel.com>
	Date:   Wed Apr 14 16:04:23 2010 -0700

	    ixgbe: fix bug with vlan strip in promsic mode

among other things added a function called ixgbe_vlan_filter_enable.
This new function wants to access and set some rx_ring parameters, but
adapter->rx_ring has already been freed.  This simply moves the free
until after the access and makes __ixgbe_shutdown look more like
ixgbe_remove.

Signed-off-by: Andy Gospodarek <andy@greyhouse.net>
Acked-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Emil Tantilov <emil.s.tantilov@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---
 drivers/net/ixgbe/ixgbe_main.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: linux-2.6.34-openSUSE-11.3/drivers/net/ixgbe/ixgbe_main.c
===================================================================
--- linux-2.6.34-openSUSE-11.3.orig/drivers/net/ixgbe/ixgbe_main.c
+++ linux-2.6.34-openSUSE-11.3/drivers/net/ixgbe/ixgbe_main.c
@@ -4765,7 +4765,6 @@ static int __ixgbe_shutdown(struct pci_d
 		ixgbe_free_all_tx_resources(adapter);
 		ixgbe_free_all_rx_resources(adapter);
 	}
-	ixgbe_clear_interrupt_scheme(adapter);
 
 #ifdef CONFIG_PM
 	retval = pci_save_state(pdev);
@@ -4800,6 +4799,8 @@ static int __ixgbe_shutdown(struct pci_d
 
 	*enable_wake = !!wufc;
 
+	ixgbe_clear_interrupt_scheme(adapter);
+
 	ixgbe_release_hw_control(adapter);
 
 	pci_disable_device(pdev);
